modules/ROOT/pages/apikit-workflow-convert-existing.adoc

```diff
@@ -1,104 +1,239 @@
-
-. Right-click the name of the RAML in /src/main/resources/api and select Validate. Proceed if there are no problems with the RAML.
-
-. Create any dependent files, or drag and drop them into Studio beside the main RAML. For example, drag the user-example.json to /src/main/resources/api.
+If you choose a backend-first design approach for Mule projects and applications that contains business logic, you can convert them into an APIkit project and expose them as REST services. Use one of these approaches:
+ 
+* xref:apikit-workflow-convert-existing.adoc#manual-approach[]
+* xref:apikit-workflow-convert-existing.adoc#scaffolding-approach[]
+
+Consider the size and complexity of the application when deciding on the best approach.
+
+The examples for these approaches are based on the following Mule application that retrieves users from a database:
+
+[source,xml,linenums]
+----
+<mule>
+    <http:listener-config name="HTTP_Listener_config">
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+	<db:config name="Database_Config">
+		<db:my-sql-connection host="localhost" port="3306" user="admin" password="admin" database="usersdb" />
+	</db:config>
+	<flow name="GET-Users-FLOW">
+		<http:listener config-ref="HTTP_Listener_config" path="/users"/>
+		<db:select config-ref="Database_Config">
+			<db:sql ><![CDATA[SELECT * FROM USERS]]></db:sql>
+		</db:select>
+	</flow>
+	<flow name="GET-User-By-Id-FLOW">
+		<http:listener config-ref="HTTP_Listener_config" path="/userById" />
+		<db:select config-ref="Database_Config" >
+			<db:sql ><![CDATA[SELECT * FROM USERS WHERE :userIdParameter == userID]]></db:sql>
+			<db:input-parameters ><![CDATA[#[{userIdParameter: attributes.queryParams.id}]]]></db:input-parameters>
+		</db:select>
+	</flow>
+</mule>
+----
+
+== Before You Begin
+
+To define the API specification for your service, see xref:apikit-4-implement-rest-api.adoc#design-your-api-specification[Design Your API Specification at Building, Implementing, and Testing a REST API]. To illustrate this example, create or copy the API specification and JSON example used in that workflow to the `/src/main/resources/api` folder.
+
+== Manual Approach
+
+Follow these steps to configure a functional router for your existing flows manually.
+
+=== Step 1: Create a Main Flow
+
+Within the APIkit project, create a main flow that serves as the entry point:
+
+. Drag the following components from the Mule palette to the canvas:
 +
-Ignore error messages in the Problems window that pertain to undefined attributes.
-
-== Create a Main Flow
-
-Within the APIkit project, build a flow.
-
-. Drag the following components from the Mule palette:
+* An *HTTP Listener* source as a flow source
+* An *APIkit Router* operation within the flow process
 +
-* An HTTP Listener
-* An APIkit Router
+. In the properties editor for *HTTP Listener* in *Connector Configuration*, select the existing `HTTP_Listener_Config`.
```

Comment:
Is HTTP_Listener_Config how this appears in the properties editor? If so,

...select *HTTP_Listener_Config*.

Reply:
Yes! Thanks!